### PR TITLE
Post/Patch Trigger APIs

### PR DESCRIPTION
**Pull request by yongpingchen**
### Description
Since the required fields  for creating trigger and updating trigger are different. It will be clear to provide seperate class for each operation(create or update). 

#### Change
- make TriggerCommandObject.targetID optional
  - PostCommandTrigger will use resource thing by default.
  - PathCommandTrigger will not update if it is null.

#### Deprecated
- CommandTriggerRequest
- ServerCodeTriggerRequest

#### New class
- TrggerCommandObject: used as a property in PostCommandTriggerRequest and PatchCommandTriggerRequest.
- PostCommandTriggerRequest: used in ThingIFAPI/APIAuthor.PostCommandTrigger
- PatchCommandTriggerRequest: used in ThingIFAPI/APIAuthor.PatchCommandTrigger
- PostServercodeTriggerRequest: used in ThingIFAPI/APIAuthor.PostServercodeTrigger
- PatchServercodeTriggerRequest: used in ThingIFAPI/APIAuthor.PatchServercodeTrigger

### CODE REVIEW

**Comment by yongpingchen** [comment]
API doc of last build https://435-61766435-gh.circle-artifacts.com/0//tmp/circle-artifacts.sLjRLy2/html/index.html

**Comment by yongpingchen** [comment]
@shinsuke-nara  can you review it?

**Comment by satoshikumano** [comment]
CommandTriggerCommandObject is bit lengthy name to me.

**Comment by yongpingchen** [comment]
what about TriggerCommandObject?

**Comment by yongpingchen** [comment]
@nfukuzaki can you review the PR for API design?

File: src/RequestObjects.ts [hunk 1]
```diff
@@ -239,54 +239,138 @@ export class ListQueryOptions {
...
 /**
- * Represents the request for creating/updating a server code trigger.
+ * Represents the request for creating a server code trigger.
  * @prop {ServerCode} serverCode Details of the server code to execute.
  * @prop {Predicate} predicate Predicate of the condition met for the trigger to execute.
  */
-export class ServerCodeTriggerRequest{
+export class PostServerCodeTriggerRequest{
     public serverCode: ServerCode;
     public predicate: Predicate;
```
Comment (yongpingchen) [inline src/RequestObjects.ts]: Add missing title, description, metadata

**Comment by yongpingchen** [comment]
@satoshikumano I didn't implementation builder method yet. Please confirm.

**Pull request merged**